.github/actions/setup-environment/action.yml

```
@@ -0,0 +1,78 @@
+name: 'Setup Environment'
+description: 'Setup development environment for Folo project'
+inputs:
+  cache-key-suffix:
+    description: 'Suffix for cache keys to differentiate between workflows'
+    required: true
+
+runs:
+  using: 'composite'
+  steps:
+    - name: Install system dependencies (Linux)
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        sudo apt update
+        sudo apt install -y git git-lfs build-essential cmake gcc make curl
+        # Set up Git LFS
+        git lfs install
+    
+    - name: Install PowerShell 7 (Linux)
+      if: runner.os == 'Linux'
+      shell: bash
+      run: |
+        # Download and install PowerShell 7
+        wget -q "https://packages.microsoft.com/config/ubuntu/$(lsb_release -rs)/packages-microsoft-prod.deb"
+        sudo dpkg -i packages-microsoft-prod.deb
+        sudo apt update
+        sudo apt install -y powershell
+        # Verify PowerShell installation
+        pwsh -c "Write-Host 'PowerShell installed successfully'"
+    
+    - name: Install Rust toolchain
+      uses: dtolnay/rust-toolchain@stable
+    
+    - name: Install nightly Rust toolchain
+      uses: dtolnay/rust-toolchain@nightly
+      with:
+        components: miri, rustfmt
+    
+    - name: Setup Rust cache
+      uses: Swatinem/rust-cache@v2
+      with:
+        # Cache even on build failures to speed up subsequent runs
```

Caching on failures feels risky. What if the failure was because a tool installation silently failed and corrupted the cache? Dubious benefits while increasing risk - let's remove this unless we have a specific problem with build performance (which I think we do not).